src/engine/tt_hash_table.h

```diff
@@ -106,13 +106,27 @@ struct TtHashEntry {
 
 class TtHashTable {
 public:
-    static void clear()
+    /* frees the memory of the current table (if any) and allocates the amount provided in MB
+     * NOTE: NOT THREAD SAFE */
+    static void setSizeMb(std::size_t sizeMb)
     {
+        if (s_ttHashSize > 0) {
+            std::free(s_ttHashTable);
+        }
 
-        for (auto& entry : s_ttHashTable) {
-            entry.key = 0;
-            entry.data = TtHashEntryData();
+        s_ttHashSize = tableSizeFromMb(sizeMb);
+        s_ttHashTable = static_cast<TtHashEntry*>(std::aligned_alloc(alignof(TtHashEntry), s_ttHashSize * sizeof(TtHashEntry)));
```

Comment:
Are we making a considered exception to our 'no heap allocation' guideline here? I think it makes sense, but I'd like to check we're still on the same page about the guidelines :)

Reply:
For the hash table we don't have much of a choice. I has to be scalable (tournament rules etc).
But for the sake of performance, it should really not matter here.
The guidelines are for the engine itself (searcher) where runtime heap allocations will be super slow compared to stack allocations :)